File: backend/app/schemas/work_rule.py
# WorkRule 관련 스키마
from datetime import datetime, date
from typing import Literal, Optional
from pydantic import BaseModel, Field
import uuid


class WorkRuleSectionSchema(BaseModel):
    """취업규칙 섹션"""
    section_number: int = Field(..., ge=1, le=14)
    title: str = Field(..., min_length=1, max_length=200)
    content_html: str = Field(..., min_length=1)
    is_required: bool = True
    law_reference: Optional[str] = None


class WorkRuleContentSchema(BaseModel):
    """취업규칙 컨텐츠"""
    sections: list[WorkRuleSectionSchema]


class WorkRuleCreate(BaseModel):
    """취업규칙 생성 요청"""
    industry_type: Literal["manufacturing", "food_service", "service", "it"] = Field(...)
    effective_date: Optional[date] = None


class WorkRuleUpdate(BaseModel):
    """취업규칙 수정 요청"""
    content: Optional[WorkRuleContentSchema] = None
    effective_date: Optional[date] = None
    status: Optional[Literal["draft", "under_review", "active"]] = None
    worker_consent_count: Optional[int] = Field(None, ge=0)
    total_worker_count: Optional[int] = Field(None, ge=0)
    approval_date: Optional[date] = None


class WorkRuleGenerateRequest(BaseModel):
    """AI 초안 생성 요청"""
    industry_type: Optional[Literal["manufacturing", "food_service", "service", "it"]] = None
    additional_context: Optional[str] = Field(None, max_length=1000)


class WorkRuleReviseRequest(BaseModel):
    """개정 요청"""
    revision_reason: str = Field(..., min_length=1, max_length=500)
    effective_date: Optional[date] = None


class WorkRuleResponse(BaseModel):
    """취업규칙 응답"""
    id: uuid.UUID
    company_id: uuid.UUID
    version: int
    status: str
    industry_type: str
    content: dict
    effective_date: Optional[date] = None
    approval_date: Optional[date] = None
    worker_consent_count: Optional[int] = None
    total_worker_count: Optional[int] = None
    revision_reason: Optional[str] = None
    ai_generated: bool
    ai_model: Optional[str] = None
    docx_url: Optional[str] = None
    pdf_url: Optional[str] = None
    filed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class WorkRuleListItem(BaseModel):
    """취업규칙 목록 아이템"""
    id: uuid.UUID
    version: int
    status: str
    industry_type: str
    effective_date: Optional[date] = None
    approval_date: Optional[date] = None
    worker_consent_count: Optional[int] = None
    ai_generated: bool
    filed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DownloadResponse(BaseModel):
    """다운로드 응답"""
    download_url: str
    filename: str
    expires_at: str


class ConsentChecklistStep(BaseModel):
    """동의 절차 체크리스트 단계"""
    step: int
    title: str
    description: str
    law_reference: str
    is_required: bool


class ConsentChecklistResponse(BaseModel):
    """동의 절차 체크리스트 응답"""
    checklist: list[ConsentChecklistStep]
    employee_count: int
    consent_threshold: int
    consent_type: str  # "majority" | "opinion"


class TemplateSection(BaseModel):
    """템플릿 섹션"""
    section_number: int
    title: str
    description: str


class TemplateResponse(BaseModel):
    """템플릿 응답"""
    industry_type: str
    industry_name: str
    description: str
    sections: list[TemplateSection]

File: frontend/types/work-rule.ts
/**
 * 취업규칙 관련 타입 정의
 */

// 취업규칙 상태
export type WorkRuleStatus = 'draft' | 'under_review' | 'active' | 'superseded';

// 업종 타입 (설계서 기준)
export type IndustryType = 'manufacturing' | 'food_service' | 'service' | 'it';

// 취업규칙 섹션
export interface WorkRuleSection {
  section_number: number;
  title: string;
  content_html: string;
  is_required: boolean;
  law_reference?: string;
}

// 취업규칙 내용
export interface WorkRuleContent {
  sections: WorkRuleSection[];
}

// 취업규칙 기본 정보
export interface WorkRule {
  id: string;
  company_id: string;
  version: number;
  status: WorkRuleStatus;
  industry_type: string;
  content: WorkRuleContent;
  effective_date?: string;
  approval_date?: string;
  worker_consent_count?: number;
  total_worker_count?: number;
  revision_reason?: string;
  ai_generated: boolean;
  ai_model?: string;
  docx_url?: string;
  pdf_url?: string;
  filed_at?: string;
  created_at: string;
  updated_at: string;
}

// 취업규칙 목록 아이템 (축약된 정보)
export interface WorkRuleListItem {
  id: string;
  version: number;
  status: WorkRuleStatus;
  industry_type: string;
  effective_date?: string;
  approval_date?: string;
  worker_consent_count?: number;
  ai_generated: boolean;
  filed_at?: string;
  created_at: string;
  updated_at: string;
}

// 취업규칙 생성 요청
export interface WorkRuleCreate {
  industry_type: IndustryType;
  effective_date?: string;
}

// 취업규칙 업데이트 요청
export interface WorkRuleUpdate {
  content?: WorkRuleContent;
  effective_date?: string;
  status?: WorkRuleStatus;
  worker_consent_count?: number;
  total_worker_count?: number;
  approval_date?: string;
}

// AI 초안 생성 요청
export interface WorkRuleGenerateRequest {
  industry_type?: IndustryType;
  additional_context?: string;
}

// 취업규칙 개정 요청
export interface WorkRuleReviseRequest {
  revision_reason: string;
  effective_date?: string;
}

// 다운로드 응답
export interface DownloadResponse {
  download_url: string;
  filename: string;
  expires_at: string;
}

// 동의 절차 체크리스트 스텝
export interface ConsentChecklistStep {
  step: number;
  title: string;
  description: string;
  law_reference: string;
  is_required: boolean;
}

// 동의 절차 체크리스트 응답
export interface ConsentChecklistResponse {
  checklist: ConsentChecklistStep[];
  employee_count: number;
  consent_threshold: number;
  consent_type: string;
}

// 템플릿 섹션
export interface TemplateSection {
  section_number: number;
  title: string;
  description: string;
}

// 템플릿 응답
export interface TemplateResponse {
  industry_type: string;
  industry_name: string;
  description: string;
  sections: TemplateSection[];
}

// 업종 옵션
export const INDUSTRY_OPTIONS: Array<{ value: IndustryType; label: string }> = [
  { value: 'manufacturing', label: '제조업' },
  { value: 'food_service', label: '요식업' },
  { value: 'service', label: '서비스업' },
  { value: 'it', label: 'IT/정보통신' },
];

// 상태 라벨
export const STATUS_LABELS: Record<WorkRuleStatus, string> = {
  draft: '작성중',
  under_review: '검토중',
  active: '시행중',
  superseded: '폐지됨',
};

// 업종 라벨 변환 함수
export function getIndustryLabel(type: string): string {
  const option = INDUSTRY_OPTIONS.find(opt => opt.value === type as IndustryType);
  return option?.label || type;
}

// 상태 라벨 변환 함수
export function getStatusLabel(status: WorkRuleStatus): string {
  return STATUS_LABELS[status] || status;
}
